enhancements/ingress/aws-lb-eip-selection.md

```diff
@@ -0,0 +1,443 @@
+---
+title: aws-lb-eip-selection
+authors:
+- "@miheer"
+reviewers:
+- "@gcs278"
+- "@alebedev87"
+- "@Miciah"
+- "@mtulio"
+- "@candita"
+- "@frobware"
+- "@rfredette"
+approvers:
+- "@patrickdillon"
+api-approvers:
+- None
+creation-date: 2024-05-29
+last-updated: 2024-09-04
+tracking-link:
+  - https://issues.redhat.com/browse/CORS-3440
+see-also:
+  - "enhancements/ingress/lb-subnet-selection-aws.md"
+replaces:
+  - "enhancements/installer/aws-customer-provided-subnets.md"
+superseded-by:
+---
+
+# Installer LoadBalancer EIP Selection for AWS
+
+## Summary
+
+This enhancement extends the OpenShift Installer's install-config, enabling cluster admins to
+configure EIPs for AWS NLB load balancer created for their default NLB IngressController at install time.
+This proposal allows the install-time configuration of subnets for the `default` IngressController.
+
+## Definitions and Terminology
+
+### The `default` IngressController
+
+This proposal will refer to the IngressController that gets created automatically during installation and handles
+the platform routes (console, auth, canary, etc.) as the `default` IngressController.
+
+### CCM 
+CCM refers to Cloud Controller Manager which calls [AWS LoadBalancer Controller](https://github.com/kubernetes-sigs/aws-load-balancer-controller) to create AWS LBs.
+
+### Some more terminologies
+EIP = AWS Elastic IP, NLB = AWS Network LoadBalancer, AZ = Availability Zone, LB = LoadBalancer
+
+
+## Motivation
+
+### User Stories
+- As a cluster administrator using installer, I want to configure default NLB IngressController to use EIPs.
+- As a cluster administrator of OpenShift on AWS (ROSA) using installer, I want to use static IPs (and therefore AWS Elastic IPs) so that
+  I can configure appropriate firewall rules.
+  I want the default AWS Load Balancer that they use (NLB) for their router to use these EIPs.
+- As a cluster administrator who has purchased a block of public IPv4 addresses, I want to use these IP addresses, so
+  that I can avoid Amazon's new charge for public IP addresses.
+
+### Goals
+- Users are able to use EIPs for a default NLB `IngressController` at install time.
+- Check for unassociated EIPs before passing to CCM.
```

Comment:
Think we should make this more generic? Otherwise, you are missing a couple of validations here.
```suggestion
- Add validation to the installer to prevent invalid EIP configurations
```